### PR TITLE
Halon combustion buff. Halon combustion can now release atmos resin. Atmos firefighter backpack now needs halon to refill.

### PR DESCRIPTION
**Pull request by Pickle-Coding**
## About The Pull Request
Allows halon combustion to release atmos resin. Reduces halon combustion minimum temperature to 70C.

Atmos firefighter backpack now uses halon, and needs halon to refill. The nozzle can be used on a canister or any portable atmospheric device that contains halon to refill, and will consume 1 mole of halon per unit refilled. It will also release firefighting foam when in extinguishing mode.

Halves the unit consumption of the nozzle in resin launcher mode.

## Why It's Good For The Game
Halon was underwhelming. It could not compete with the firefighter backpack or the modsuit module. Halon should be the most robust fire suppressant one can obtain. Making it great at containing and suppressing fires with resin will make halon a competitive choice. The temperature requirement got reduced so it can pre-emptively act on a potential nearby fire before it can spread.

The firefighter backpacks did quite a lot for something that only required water, so making it require halon will make halon more important if someone wants to use it extensively. The firefighter backpacks in extinguishing mode releasing water is a bit silly for something that requires halon to refill, so now it will release firefighting foam.

Now that it requires something that isn't roundstart to refill, the consumption rate has been halved for the resin launcher mode to allow more usage before needing to create halon.

## Changelog

:cl:
add: Halon combustion will now release atmos resin.
balance: Reduces halon combustion minimum temperature requirement to 70C.
balance: Atmos firefighter backpack now requires halon to refill.
balance: Atmos firefighter backpack will release firefighting foam instead of water when in extinguishing mode.
balance: Halves the consumption rate for the atmos firefighter backpack when in resin launcher mode.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
hows it work with the modsuit one

**Comment by Pickle-Coding** [comment]
Does the modsuit one use its own special nozzle or something? I haven't checked that yet.

**Comment by Pickle-Coding** [comment]
No idea why that caused hard deletes.

**Comment by Pickle-Coding** [comment]
Works with modsuit mister.

File: code/game/objects/effects/effect_system/fluid_spread/effects_foam.dm [hunk 1]
```diff
@@ -408,14 +408,14 @@
...
+		air.temperature = T20C
 		for(var/obj/effect/hotspot/fire in location)
 			qdel(fire)
 
 		var/list/gases = air.gases
 		for(var/gas_type in gases)
 			switch(gas_type)
-				if(/datum/gas/oxygen, /datum/gas/nitrogen)
+				if(/datum/gas/oxygen, /datum/gas/nitrogen, /datum/gas/halon)
 					continue
```
Comment (Fikou) [inline code/game/objects/effects/effect_system/fluid_spread/effects_foam.dm:419]: while youre touching this, this part feels really weird and i have no idea why its a switch type. can you just make it
if(gas_type in list(/datum/oxygen, etc))

**Comment by Fikou** [comment]
> Works with modsuit mister.

how do you refill it? just slap on a canister?

File: code/game/objects/effects/effect_system/fluid_spread/effects_foam.dm (hunk 1 above)
Comment (Pickle-Coding) [reply code/game/objects/effects/effect_system/fluid_spread/effects_foam.dm:419]: Yeah.

**Comment by Pickle-Coding** [comment]
Slap it on any canister that has halon in it.

**Comment by BongoDongo2000** [comment]
*inhale*
**I'm going to preface this with the following facts i believe to be true, based off my experience:**

- The majority of fires or floods are started by atmos techs/engineers
- They are the ones with access, know of, and even bother with the atmos firefighter backpack.
- Atmos techs fall into two categories: Those who fuck off and dont even setup atmos, and those who will just keep experimenting with atmos mixing because of changes to the atmos mixing code and wont go fight fires as long as atmos isn't on fire itself. The rare ones that would do are usually those who are new to atmos and don't know much.
- People rarely use the atmos firefighter backpack to fight fires

**I think Halon turning into atmos resin at 70c is a good change. HOWEVER....making it the only way to refill atmos firefighter backpacks is not.**

> Halon was underwhelming. It could not compete with the firefighter backpack or the modsuit module. Halon should be the most robust fire suppressant one can obtain. Making it great at containing and suppressing fires with resin will make halon a competitive choice. The temperature requirement got reduced so it can pre-emptively act on a potential nearby fire before it can spread.

Halon got added in and was just never used. It can't "compete" with them because there is no good way in game to effectively use it against any fire. There **IS** one way but i will bring that up later.

>The firefighter backpacks did quite a lot for something that only required water, so making it require halon will make halon more important if someone wants to use it extensively. The firefighter backpacks in extinguishing mode releasing water is a bit silly for something that requires halon to refill, so now it will release firefighting foam.

Its a specialized tool with only a couple on the station. Its rarely used and sometimes its the ONLY way to stop a plasmafire WHILE leaving a room usable.

>Now that it requires something that isn't roundstart to refill, the consumption rate has been halved for the resin launcher mode to allow more usage before needing to create halon.

Requiring Halon to refill is nothing but a nerf for the atmos firefighter backpack. People might make it to play around with for a month, then it will get ignored again. Atmosians would rather make the more fun gases than make a SUPER NICHE GAS. This will then cause atmos firefighter backpacks to become even more niche or not even used except as disposable items because no one wants to fucking spend 5-10 minute setting up and making Halon just to refill it, in the SMALL OFF CHANCE THAT THERE WILL BE FUCKING PLASMAFLOOD THEY WOULDN"T SEE COMING BECAUSE THEY"D BE IN ATMOS ALREADY AND EITHER BE THE ONES DOING THE FLOOD OR BE MURDERED BY THE PERSON WANTING TO DO IT.

But BongoDongo2000, it will be fun! Its immersive! It will reward player skill! No its adding more tedium and a timesink while taking away existing functionality for something that I think will rarely if ever get used. Remember when fermichem was implemented and PH made making chems annoying? What did people do with the "fun" and "immersive" and "skill"? They used buffer chems to get around PH turning your chems into sludge. People didnt like it and we were stuck with it for YEARS until 1-2 months ago. 

But BongoDongo2000 you're wrong! People will love this and use it! Remember gas vendors? Yeah that guy thought atmos techs would love to make gases, put them in the vendor, and then people would buy those gases. I've NEVER seen it used except to cool fusion gases. Thats it.

I could be wrong and this will somehow be amazing, but i doubt it. End rant. Now onto something shocking!

**How you can ALREADY use Halon in the game to fight fires!!!:**
Step 1: Waste whatever amount of time it takes to setup and make some Halon.
Step 2: Make a flame thrower.
Step 3: Get a plasma tank (TANK FROM THE DISPENSOR, NOT A FULL CANISTER) and use an air pump to empty it.
Step 4: Either fill another air pump with Halon or grab the canister you spend however long on making/filling with Halon.
Step 5: Use the air pump or canister to fill the plasma tank with Halon.
Step 6: Now attach the plasma tank filled with Halon to the flamethrower.

Congratulations, you now have a weapon that will shoot/vent Halon into a room or direction!

**But wait! Thats not enough for you? You realllly want to add another way to use Halon? How could we make Halon more useful while adding new and fun functionality to the game without adding tedium or nerfing anything?**

Glad you asked and I have a solution for you! Air Alarms! Yes sir, good old air alarms. Do you know how Halon was/is used against fires? Why, with fire suppression systems! But how to combine Air Alarms, Halon, and fire suppression systems?

- Add to air alarms a "Emergency Fire Suppression" button, make is usable while the air alarm is locked.
- Make the button only usable while locked if the amount of plasma/trit/hydrogen in the air is above the flammable point and the fire alarm has been triggered by, well, a fire.
- Make the button usable regardless if the air alarm is unlocked or emagged.
- Make the air alarm announce a countdown from 5 when the button is pressed and then it engages, unless emagged. 
- Add to air alarms a "Fire Suppression Gas Storage" button, make it only available when the air alarm is unlocked
- Chop the air pump code so it can only fill/empty from a canister, then slap that on the "Fire Suppression Gas Storage" button. Give it a SMALL volume but roundstart have it be filled with Halon over that volume.
- Add code to switch gas network for URINARY vents from the distribution network to the gas stored in the air alarm. Slap that on the "Emergency Fire Suppression" button. Have it vent its entire VOLUME over 5 seconds and then switch back to the distribution network and have a 2-5 minute cooldown + need a refill.
- If thats too hard, have it disable the vents and then chop the clogged scrubber event (the one that causes foam overflow) and use that instead to spread the Halon/gas.
- Add code to toggle the ability to only accept Halon gas and have that on as default BUT toggle it off so it will accept any gas if emagged.
- Add code to announce on engineering channel when/where the "Emergency Fire Suppression" button is pressed and the held/worn ID of whoever did it unless the air alarm is emagged.
- Add code to log who presses the "Emergency Fire Suppression" button and who fills the gas storage on the air alarm.
- Add an additional wire to air alarms. Cutting disables the "Emergency Fire Suppression", pulsing triggers it.

We now have a new system with opportunity for play that makes use of Halon, can be refilled with player created Halon, or emagged to "suppress" the plasma fire with even more plasma/trit/hydrogen.

**Comment by LemonInTheDark** [comment]
Retroactive, but I'm not a massive fan of this. making halon stronger sure but I don't see a good reason to handycap firefighter stuff

**Pull request closed without merging**